zng/docs/spec.md

```diff
@@ -314,7 +387,66 @@ It is an error to define an alias that has the same name as a primitive type.
 It is also an error to redefine a previously defined alias with a
 type that differs from the original definition.
 
-### 3.1.2 End-of-Stream Markers
+### 3.1.2 Compressed Value Message Block
+
+Following a header byte of `0xf6` is a compressed value message block.
+Such a block comprises a compressed sequence of value messages.  The
+sequence must not include control messages.
+
+> The reason control messages are not allowed in compressed blocks is to
+> allow for optimizations that discard entire buffers of data based on
+> heuristics to know a filtering predicate can't be true based on a
+> quick scan of the data (e.g., using the Boyer-Moore algorithm to determine
+> that a comparison with a string constant would not work for any record
+> value in the buffer).  Since blocks may be dropped without parsing using
+> such an optimization, any typedefs should be lifted out into the zng data
+> stream in front of the compressed blocks (i.e., the stream is rearranged
+// but its always safe to move typedefs earlier in the stream as long as
```

Comment:
```suggestion
// but it's always safe to move typedefs earlier in the stream as long as
```

Also, the `//` in there makes this paragraph look like it's still under construction. Not sure if it could be replaced with a simple comma or if this was some "XXX"-style placeholder to remind you to come back to it.